### PR TITLE
Chore/ruby 2769 enhance govpay logging

### PR DESCRIPTION
**Pull request by PaulDoyle-DEFRA**
https://eaflood.atlassian.net/browse/RUBY-2769

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,17 +1,30 @@
...
+  defra_ruby_style:
+    - default.yml
 
-Style/StringLiteralsInInterpolation:
-  Enabled: true
-  EnforcedStyle: double_quotes
+require:
+  - rubocop-factory_bot
+  - rubocop-rake
+  - rubocop-rspec
```
Comment (brujeo) [inline .rubocop.yml:10]: These are already included into default defra-ruby-style config
https://github.com/DEFRA/defra-ruby-style/blob/main/default.yml

File: .rubocop.yml [hunk 2]
```diff
@@ -1,17 +1,30 @@
...
 
 Layout/LineLength:
   Max: 120
 
 Metrics/BlockLength:
   Exclude:
     - 'spec/**/*'
```
Comment (brujeo) [inline .rubocop.yml:17]: These exceptions are already included into default defra-ruby-style config
https://github.com/DEFRA/defra-ruby-style/blob/main/default.yml

This is a brand new gem, so ideally we should not have anything in this config apart from

```
inherit_gem:
  defra_ruby_style:
    - default.yml
```

File: Gemfile.lock [hunk 3]
```diff
@@ -10,8 +10,11 @@ GEM
     addressable (2.8.5)
       public_suffix (>= 2.0.2, < 6.0)
     ast (2.4.2)
+    byebug (11.1.3)
     crack (0.4.5)
       rexml
+    defra_ruby_style (0.3.0)
```
Comment (brujeo) [inline Gemfile.lock:16]: update defra_ruby_style to most recent version 0.4.0 
That has all the recent changes

File: lib/defra_ruby_govpay/api.rb [hunk 4]
```diff
@@ -22,49 +22,54 @@ def initialize(config = DefraRubyGovpay.configuration)
     end
 
     def send_request(method:, path:, params: nil, is_moto: false)
-      Rails.logger.debug build_log_message(method, path, params, is_moto)
+      @is_moto = is_moto
+      DefraRubyGovpay.logger.debug DefraRubyGovpay.logger.debug build_log_message(method, path, params)
```
Comment (brujeo) [inline lib/defra_ruby_govpay/api.rb:26]: Not sure what this line suppose to do. Why DefraRubyGovpay.logger.debug added twice?
`DefraRubyGovpay.logger.debug DefraRubyGovpay.logger.debug build_log_message(method, path, params)`

Comment (PaulDoyle-DEFRA) [reply lib/defra_ruby_govpay/api.rb:26]: Copy and paste error, fixed

File: .rubocop.yml (hunk 2 above)
Comment (PaulDoyle-DEFRA) [reply .rubocop.yml:17]: OK, but `Style/OpenStructUse` is problematic so I need to keep that exclusion for now. I'll move it to `.rubocop_todo.yml` though so it's flagged as something that needs attention when we have more time.

**Review by brujeo — APPROVED** [review]

**Pull request merged**